src/easier68k/core/opcodes/move.py

```diff
@@ -0,0 +1,216 @@
+
+        to_return = bytearray.fromhex(hex(int(tr, 2))[2:])  # Convert to a bytearray
+        return to_return
+
+    def execute(self, simulator: M68K):
+        pass
+
+    def __str__(self):
+        # Makes this a bit easier to read in doctest output
+        return 'Move command: Size {}, src {}, dest {}'.format(self.size, self.src, self.dest)
+
+    @staticmethod
+    def is_valid(command: str, parameters: str) -> (bool, list):
+        """
+        Tests whether the given command is valid
+
+        >>> Move.is_valid('MOVE.B', 'D0, D1')[0]
+        True
+
+        >>> Move.is_valid('MOVE.W', 'D0')[0]
+        False
+
+        >>> Move.is_valid('MOVE.G', 'D0, D1')[0]
+        False
+
+        >>> Move.is_valid('MOVE.L', 'D0, A2')[0]
+        False
+
+        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+        :param parameters: The parameters after the command (such as the source and destination of a move)
+        :return: Whether the given command is valid and a list of issues/warnings encountered
+        """
+        issues = []
+        parts = command.split('.')  # Split the command by period to get the size of the command
+        assert len(parts) <= 2  # If we have more than 2 parts something is seriously wrong
+        if len(parts) != 1:  # Has a size specifier
+            try:
+                assert parts[1] in Move.allowed_sizes, "Size {} isn't allowed for command {}".format(parts[1], command[0])
+            except AssertionError as e:
+                issues.append((e.args[0], 'ERROR'))
+                return False, issues
+
+        # Split the parameters into EA modes
+        try:
+            params = parameters.split(',')
+            assert len(params) == 2, 'Must have two parameters'
+        except AssertionError as e:
+            issues.append((e.args[0], 'ERROR'))
+            return False, issues
+
+        try:
+            src = EAMode.parse_ea(params[0].strip())  # Parse the source and make sure it parsed right
+            assert src.mode > EAMode.ERR, 'Error parsing src'  # -1 means error
+        except AssertionError as e:
+            issues.append((e.args[0], 'ERROR'))
+            return False, issues
+
+        try:
+            dest = EAMode.parse_ea(params[1].strip())
+            assert dest.mode > EAMode.ERR, 'Error parsing dest'  # -1 means error
+        except AssertionError as e:
+            issues.append((e.args[0], 'ERROR'))
+            return False, issues
+
+        try:
+            assert src.mode != EAMode.ARD, 'Invalid addressing mode'  # Only invalid src is address register direct
+            assert dest.mode != EAMode.ARD and dest.mode != EAMode.IMM, 'Invalid addressing mode'
+        except AssertionError as e:
+            issues.append((e.args[0], 'ERROR'))
+            return False, issues
+
+        return True, issues
+
+    @staticmethod
+    def get_word_length(command: str, parameters: str) -> (int, list):
```

Comment:
maybe we can redesign this so that we aren't redoing this in each of the commands. should make each of the opcode classes as short as they can be